dbt/models/default/default.vw_pin_sale.sql

```diff
@@ -249,93 +238,192 @@ sales_val AS (
         sf.run_id AS sv_run_id,
         sf.version AS sv_version
     FROM
-        {{ source('sale', 'flag') }}
-            AS sf
+        {{ source('sale', 'flag') }} AS sf
     INNER JOIN max_version_flag AS mv
         ON sf.meta_sale_document_num = mv.meta_sale_document_num
         AND sf.version = mv.max_version
+),
+
+-- Introducing cte_sales to precompute the coalesced values
+cte_sales AS (
+    SELECT
+        COALESCE(uq_sales.pin, md_sales.pin) AS pin_coalesced,
+        CASE
+            WHEN md_sales.sale_date IS NOT NULL
```

Comment:
Let's add a comment as to why year (and date for that matter) is handled differently than the other coalesces.